core/src/main/java/org/kiwix/kiwixmobile/core/main/AddNoteDialog.kt

```diff
@@ -0,0 +1,444 @@
+      )
+    } else ""
+    // If couldn't find a dot and/or slash in the url
+  }
+
+  // Override onBackPressed() to respond to user pressing 'Back' button on navigation bar
+  override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+    return object : Dialog(requireContext(), theme) {
+      override fun onBackPressed() {
+        exitAddNoteDialog()
+      }
+    }
+  }
+
+  private fun exitAddNoteDialog() {
+    if (noteEdited) {
+      alertDialogShower.show(
+        KiwixDialog.NotesDiscardConfirmation,
+        ::dismissAddNoteDialog
+      )
+    } else {
+      // Closing unedited note dialog straightaway
+      dismissAddNoteDialog()
+    }
+  }
+
+  private fun disableMenuItems() {
+    if (toolbar.menu != null) {
+      saveItem.isEnabled = false
+      shareItem.isEnabled = false
+      deleteItem.isEnabled = false
+      saveItem.icon.alpha = DISABLE_ICON_ITEM_ALPHA
+      shareItem.icon.alpha = DISABLE_ICON_ITEM_ALPHA
+      deleteItem.icon.alpha = DISABLE_ICON_ITEM_ALPHA
+    } else {
+      Log.d(TAG, "Toolbar without inflated menu")
+    }
+  }
+
+  private fun enableSaveNoteMenuItem() {
+    if (toolbar.menu != null) {
+      saveItem.isEnabled = true
+      saveItem.icon.alpha = ENABLE_ICON_ITEM_ALPHA
+    } else {
+      Log.d(TAG, "Toolbar without inflated menu")
+    }
+  }
+
+  private fun enableDeleteNoteMenuItem() {
+    if (toolbar.menu != null) {
+      deleteItem.isEnabled = true
+      deleteItem.icon.alpha = ENABLE_ICON_ITEM_ALPHA
+    } else {
+      Log.d(TAG, "Toolbar without inflated menu")
+    }
+  }
+
+  private fun enableShareNoteMenuItem() {
+    if (toolbar.menu != null) {
+      shareItem.isEnabled = true
+      shareItem.icon.alpha = ENABLE_ICON_ITEM_ALPHA
+    } else {
+      Log.d(TAG, "Toolbar without inflated menu")
+    }
+  }
+
+  override fun onViewCreated(
+    view: View,
+    savedInstanceState: Bundle?
+  ) {
+    super.onViewCreated(view, savedInstanceState)
+    toolbar.setTitle(R.string.note)
+    toolbar.setNavigationIcon(R.drawable.ic_close_white_24dp)
+    toolbar.setNavigationOnClickListener {
+      exitAddNoteDialog()
+      closeKeyboard()
+    }
+    toolbar.setOnMenuItemClickListener { item: MenuItem ->
+      when (item.itemId) {
+        R.id.share_note -> shareNote()
+        R.id.save_note -> saveNote(add_note_edit_text.text.toString())
+        R.id.delete_note -> deleteNote()
+      }
+      true
+    }
+    toolbar.inflateMenu(R.menu.menu_add_note_dialog)
+    // 'Share' disabled for empty notes, 'Save' disabled for unedited notes
+    disableMenuItems()
+    add_note_text_view.text = articleTitle
+
+    // Show the previously saved note if it exists
+    displayNote()
+    add_note_edit_text.addTextChangedListener(object : TextWatcher {
+      @SuppressWarnings("EmptyFunctionBlock")
+      override fun beforeTextChanged(
+        s: CharSequence,
+        start: Int,
+        count: Int,
+        after: Int
+      ) {
+      }
```

Comment:
What this function with empty body does?

Reply:
Nothing, just what you are forced to implement as part of TextWatcher but actually we do have a class to make this conciser, SimpleTextWatcher. That should be used here instead.

Reply:
@macgills Is it `SimpleTextListener` class?

Reply:
Yup, sorry misremembered

Reply:
@macgills `SimpleTextListener` is a Subtype of `OnQueryTextListener`.How to add this on EditText?

Reply:
very good point, I guess this is the only TextWatcher in the project, how surprising.

Wanna write a `SimpleTextWatcher`? I can take it or leave it.

Reply:
sure. I want to write it.